src/pkg/services/m365/api/drive.go

```diff
@@ -123,21 +124,44 @@ func (c Drives) NewItemContentUpload(
 	return r, nil
 }
 
-const itemChildrenRawURLFmt = "https://graph.microsoft.com/v1.0/drives/%s/items/%s/children"
+//nolint:lll
+const itemChildrenRawURLFmt = "https://graph.microsoft.com/v1.0/drives/%s/items/%s/children?@microsoft.graph.conflictBehavior=%s"
+
+const (
+	conflictBehaviorFail    = "fail"
+	conflictBehaviorRename  = "rename"
+	conflictBehaviorReplace = "replace"
+)
 
 // PostItemInContainer creates a new item in the specified folder
 func (c Drives) PostItemInContainer(
 	ctx context.Context,
 	driveID, parentFolderID string,
 	newItem models.DriveItemable,
+	onCollision control.CollisionPolicy,
 ) (models.DriveItemable, error) {
+	// graph api has no policy for Skip; instead we catch
+	// and ignore the same-name failure.
```

Comment:
Ignoring seems to be done at a higher layer

```suggestion
	// and wrap the same-name failure.
```

Reply:
ah, good catch, thanks.